plugins/ltac/rewrite.ml

```diff
@@ -1476,7 +1481,8 @@ let cl_rewrite_clause_aux ?(abs=None) strat env avoid sigma concl is_hyp : resul
                 Termops.pr_evar_info env evars (Evd.find evars ev)))
         cstrs
     in
-    let newt = res.rew_to in
+    let subst = [do_subrelation_id, fst (do_subrelation_decl env res.rew_evars prop)] in
```

Comment:
Are we sure that we're going to unify the universe instances of the freshly generated constant in universe-poly mode? Before we only had one but now we get one for the let-bound variable and one for the substituted term, which might lead to unification failures.

Reply:
Good question.

The universes for `apply_subrelation` and `do_subrelation` were thrown before, but this is defined in `Prop` (`CoInductive apply_subrelation : Prop := do_subrelation.` in both `Morphisms.v` and `CMorphisms.v`), so universes seem to not matter. If confirmed, this is a worth a remark though.